servant-docs/src/Servant/Docs/Internal.hs

```diff
@@ -839,6 +845,33 @@ markdownWith RenderingOptions{..}  api = unlines $
                   xs        ->
                     formatBodies _responseExamples xs
 
+        curlStr :: Endpoint -> [HTTP.Header] -> [(Text, M.MediaType, ByteString)] -> String -> [String]
+        curlStr endpoint hdrs reqBodies basePath =
+          [  "### Sample Request:"
+          , ""
+          , "```bash"
+          , "curl -X" ++ BSC.unpack (endpoint ^. method) ++ " \\"
+          ] <>
+          maybe [] pure mbMediaTypeStr <>
+          headersStrs <>
+          maybe [] pure mbReqBodyStr <>
+          [  "  " ++ basePath ++ showPath (endpoint ^. path)
+          , "```"
+          , ""
+          ]
+
+          where mbReqBody = listToMaybe reqBodies
+                mbMediaTypeStr = mkMediaTypeStr <$> mbReqBody
+                headersStrs = mkHeaderStr <$> hdrs
+                mbReqBodyStr = mkReqBodyStr <$> mbReqBody
+                mkMediaTypeStr (_, media_type, _) =
+                  "  -H 'Content-Type: " ++ show media_type ++ " '\\"
+                mkHeaderStr (hdrName, hdrVal) =
+                  "  -H '" ++ cs (CI.original hdrName) ++ ": " ++
+                  cs hdrVal ++ "' \\"
+                mkReqBodyStr (_, _, body) = "  -d " ++ cs body ++ " \\"
```

Comment:
Will this correctly escape the request body in case it contains `"`s?

Reply:
That's a great point. Same with the headers. However I think it's up to the implementor to escape those quotes in the `ToSample` instance. Do you agree?

Reply:
Hmm, well proper escaping will take place for whatever content types are relevant for the request/response bodies (e.g `"`s are escaped in sample JSON bodies), so I would be inclined to say that the escaping would be fine. But since you're more familiar with this patch than myself for obvious reasons, I was essentially asking for your confirmation. It simply occurred to me that your patch was introducing bash escaping to the mix.

And indeed the same question stands for headers.

(I don't think relying on each sample doing things right is the best way out of this question though.)

Reply:
I don't see why the body would be different in this case than in the case where an actual request was being sent, meaning that I expect that quotes would be escaped correctly.

I tested this out, and in fact it doesn't quite do the correct thing; with a (pretty) JSON body, it did something like this:

```bash
  -d {
      "foo": "bar"
  } \
```

So I think we should actually wrap it in single quotes - then we won't have to worry about escaping `"` anyway. I think I was fooled by the `Greet` example because it was a JSON string.

Reply:
I think this should be good now

Reply:
If we wrap everything up in single quote, are we sure things will go smoothly in the presence of single quotes inside e.g the request body?

Reply:
I don't think so. If there are single quotes inside a JSON request body, they'll show up in the snippet unescaped.

Reply:
Hmm, is there something we can do to guarantee that people will be able to copy/paste any curl commands produced by this code and have it work? This is really the last blocker for me, for this PR, which is otherwise ready to land. Sorry to be a bit of a PITA here, but I'm just thinking about the scenario where in a professional setting people start generating docs with this but other folks in the company would complain about commands not working, etc.

Reply:
I guess I could use double quotes and escape double quotes in the block. I'm always wary of doing these things by hand but I'll give it a try

Reply:
Okay, done, and greet.md updated.